java-checks/src/main/java/org/sonar/java/checks/MutableMembersUsageCheck.java

```diff
@@ -61,7 +61,9 @@
   private static final MethodMatchers UNMODIFIABLE_COLLECTION_CALL = MethodMatchers.or(
     MethodMatchers.create().ofType(type -> MutableMembersUsageCheck.containsImmutableLikeTerm(type.name())).anyName().withAnyParameters().build(),
     MethodMatchers.create().ofAnyType().name(MutableMembersUsageCheck::containsImmutableLikeTerm).withAnyParameters().build(),
-    MethodMatchers.create().ofTypes("java.util.Collections").name(name -> name.startsWith("unmodifiable") || name.startsWith("singleton")).withAnyParameters().build(),
```

Comment:
Note: `name.startsWith("unmodifiable")` was useless because of the previous `containsImmutableLikeTerm` check.